pyproject.toml

```diff
@@ -24,6 +24,7 @@ select = [
 #    "D",  # see: https://pypi.org/project/pydocstyle
 #    "N",  # see: https://pypi.org/project/pep8-naming
 #    "S",  # see: https://pypi.org/project/flake8-bandit
+    "I",  # see: https://pypi.org/project/isort/
```

Comment:
this is the main change; the rest is just apply pre-commit :)
ref: https://github.com/microsoft/FLAML/issues/1252#issuecomment-1927245023